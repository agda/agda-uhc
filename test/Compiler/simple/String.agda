module String where

open import Common.IO
open import Common.List
open import Common.String
open import Common.Unit


testString : String
testString = "To boldly go where no man gone before"

printList : {A : Set} -> (A -> IO Unit) -> List A -> IO Unit
printList p [] = return unit
printList p (x âˆ· xs) =
    p x ,,
    printList p xs

main : IO Unit
main =
    putStrLn testString ,,
    printList printChar (stringToList testString) ,,
    putStrLn "" ,,
    putStrLn (fromList (stringToList testString)) ,,
    putStrLn (primShowString testString) ,,
    return unit
