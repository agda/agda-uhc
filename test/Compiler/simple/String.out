EXECUTED_PROGRAM

ret > ExitSuccess
out > To boldly go where no man gone before
out > To boldly go where no man gone before
out > To boldly go where no man gone before
out > To boldly go where no man gone before
out >
